src/test/java/org/graylog2/gelfclient/GelfMessageBuilderTest.java

```diff
@@ -84,8 +87,20 @@ public void testMissingVersionThrowsException() throws Exception {
         new GelfMessageBuilder("Test", "localhost", null).build();
     }
 
-    @Test(expectedExceptions = IllegalArgumentException.class)
-    public void testMissingLevelThrowsException() throws Exception {
-        new GelfMessageBuilder("Test").level(null).build();
+    @Test
+    public void testWithNullLevel() throws Exception {
+        final GelfMessageBuilder builder = new GelfMessageBuilder("Test").level(null);
+
+        boolean error = false;
+        final GelfMessage message;
+        try {
+            message = builder.build();
+
+            assertNull(message.getLevel());
+        } catch (Exception e) {
+            error = true;
+        }
+
+        assertFalse(error, "Null level should not throw an error");
```

Comment:
The `try-catch` block above and this assertion are unnecessary since the test will fail anyway if an exception has been thrown.